rhealpixdggs/rhp_wrappers.py

```diff
@@ -0,0 +1,537 @@
+
+    # Just return the opposite cell if k is beyond what the resolution can do
+    if k > half_circle:
+        cell = _mirror_cell_on_cube(cell)
+        return [_suid_to_str(cell.suid)]
+
+    # Init the ring and directions
+    ring = []
+    directions = ["right", "down", "left", "up"]
+
+    # Top-level cells (cube faces) are special
+    if len(rhpindex) == 1:
+        for direction in directions:
+            ring.append(cell.neighbor(direction).suid[0])
+
+    # Start in the upper left corner of the ring: it's k times left and k times up
+    else:
+        # Mapping to detect direction changes
+        direction_inverse = {
+            "right": "left",
+            "down": "up",
+            "left": "right",
+            "up": "down",
+        }
+
+        # Initialise iteration parameters
+        k_eff, max_steps, cell = _cell_ring_setup(cell, half_circle / 2, k)
+
+        # We're done if k_eff takes us all the way to the opposite cell (shouldn't happen at this point...)
+        if k_eff < 1:
+            ring.append(_suid_to_str(cell.suid))
+
+        # We have to do the full walk around the ring
+        else:
+            # Set starting point
+            cell, direction, n_steps = _find_cell_ring_start(
+                cell, k_eff, max_steps, directions, direction_inverse
+            )
+
+            # Walk around the ring one side at a time and collect cell addresses
+            for _ in range(0, len(directions)):
+                step = 0
+                while step < n_steps:
+                    # Add index to ring, take a step
+                    ring.append(_suid_to_str(cell.suid))
+                    next = cell.neighbor(direction)
+
+                    # Looking back not being the same as looking ahead means we need to realign
+                    if next.neighbor(direction_inverse[direction]) != cell:
+                        direction = direction_inverse[_neighbor_direction(next, cell)]
+
+                    # Take the step
+                    cell = next
+                    step = step + 1
+
+                # Prepare walking direction for next ring side
+                if n_steps == 2 * k_eff:
+                    direction = directions[
+                        (directions.index(direction) + 1) % len(directions)
+                    ]
+
+                # Reset number of steps along a side
+                n_steps = max_steps
+
+    return ring
+
+
+# TODO: placeholder, find out if rhp needs that function
+# def weighted_cell_ring(rphindex: str, k: int = 1) -> list[str]:
+#    pass
+
+
+def k_ring(rhpindex: str, k: int = 1, verbose=True) -> list[str]:
+    """
+    Returns the k-ring of cell indices around rhpindex at distance k (or None if rhpindex is invalid).
+
+    Also returns None if k < 0.
+
+    TODO: give the option to select another predefined DGGS, or pass in a custom one
+    """
+    if verbose:
+        print(str.format(CELL_RING_WARNING, "k"))
```

Comment:
Ditto https://docs.python.org/3/library/warnings.html#warnings.warn

Reply:
Changed this here too.